scripts/ai_generation/converter.py

```diff
@@ -0,0 +1,461 @@
+                            formatted_permissions_dict_str += f"\n      - {permission}"
+
+                dict_string = get_faq(azure_doc_text)
+                if not dict_string.startswith('{'):
+                    dict_string = '{' + dict_string
+                if not dict_string.endswith('}'):
+                    dict_string = dict_string + '}'
+                faq_dict = eval(dict_string)
+                
+                azure_doc_text += f"""\n<details>\n<summary><h2>FAQs</h2></summary>\n\n#### Q. What is the command-specific breakdown of permissions needed to implement this doc? \n\nA. _Format: Commands as they appears in the doc | list of unique permissions needed to run each of those commands_\n\n{formatted_permissions_dict_str}"""
+
+                if len(faq_dict) > 0:
+                    for question, answer in faq_dict.items():
+                        azure_doc_text += f"""\n\n#### Q. {question} \n\nA. {answer}\n"""
+                    azure_doc_text += f"""\n</details>"""
+                else:
+                    azure_doc_text += f"""\n</details>"""
+
+                if len(env_var_dict) > 0:
+                    for key in env_var_dict:
+                        if 'name' in key.lower() or 'resourcegroup' in key.lower():
+                            env_var_dict[key] += '$RANDOM_ID'
+                    azure_doc_text = insert_env_var_section(azure_doc_text, env_var_dict)
+
+                all_permissions = []
+                for value in permissions_dict.values():
+                    if isinstance(value, list):
+                        all_permissions.extend(value)
+                all_permissions = list(set(all_permissions))
+                all_permissions = ', '.join(all_permissions)
+
+                azure_doc_text = modify_header_permissions_ie_tag(azure_doc_text, all_permissions)
+                azure_doc_text = remove_clean_up_resources_section(azure_doc_text)
+
+                with open(os.path.join(AI_Docs, azure_doc_name.replace('.md', ''), f'README.md'), 'w') as f:
+                    f.write(azure_doc_text)
+
+if test_exec_docs != "None":
+    for exec_doc in test_exec_docs:
+        for dirpath, dirnames, filenames in os.walk(os.path.join(AI_Docs, exec_doc)):
+            try:        
+                for filename in filenames:
+                    if '.md' in filename:
+                        file_path = os.path.join(dirpath, filename)
+                        ie_test_command = f'./bin/ie test {file_path}'
+                        ie_test_result = subprocess.run(ie_test_command, shell=True)
+                        if 'Error' in ie_test_result.stdout.decode() or ie_test_result.returncode != 0:
+                            print(f"IE Log:\n{get_latest_error_log()}")
+                        else:
+                            create_pr(f"{dirpath.split('/')[-1]}-{filename}")
+            except:
+                print(f"IE Log:\n{get_latest_error_log()}")
+
+if test_exec_docs == "None":
```

Comment:
Are you able to just use an `else` statement here? What are the options for `test_exec_docs`

Reply:
changed this already since here